snuba/cli/rust_consumer.py

```diff
@@ -69,3 +130,175 @@ def rust_consumer(*, storage_names: Sequence[str], log_level: str) -> None:
+
+@dataclass(frozen=True)
+class StorageConfig:
+    name: str
+    clickhouse_cluster: ClickhouseClusterConfig
+
+
+@dataclass(frozen=True)
+class RustConsumerConfig:
+    """
+    Already resolved configuration for the Rust consumer
+    """
+
+    storages: Sequence[StorageConfig]
+    raw_topic: str
+    commit_log_topic: Optional[str]
+    replacements_topic: Optional[str]
+
+
+def resolve_consumer_config(
+    *,
+    storage_names: Sequence[str],
+    raw_topic: Optional[str],
+    commit_log_topic: Optional[str],
+    replacements_topic: Optional[str],
+    bootstrap_servers: Sequence[str],
+    commit_log_bootstrap_servers: Sequence[str],
+    replacement_bootstrap_servers: Sequence[str],
+    slice_id: Optional[int],
+) -> RustConsumerConfig:
+    """
+    Resolves the ClickHouse cluster and Kafka brokers, and the physical topic name
+    to be passed to the Rust consumer. We don't want to duplicate the cluster
+    resolution code in Rust.
+    """
+
+    storages = {
+        storage_name: get_writable_storage(StorageKey(storage_name))
+        for storage_name in storage_names
+    }
+
+    validate_storages([*storages.values()])
+
+    stream_loader = storages[storage_names[0]].get_table_writer().get_stream_loader()
+
+    resolved_raw_topic: str
+    if raw_topic is not None:
+        resolved_raw_topic = raw_topic
+    else:
+        resolved_raw_topic = (
+            stream_loader.get_default_topic_spec().get_physical_topic_name(slice_id)
+        )
+
+    resolved_commit_log_topic: Optional[str]
+    commit_log_topic_spec = stream_loader.get_commit_log_topic_spec()
+
+    if commit_log_topic_spec is None:
+        if commit_log_topic is not None:
+            raise ValueError("Commit log not supported for this storage")
+        resolved_commit_log_topic = None
+    elif commit_log_topic is not None:
+        resolved_commit_log_topic = commit_log_topic
+    else:
+        resolved_commit_log_topic = commit_log_topic_spec.get_physical_topic_name(
+            slice_id
+        )
+
+    resolved_replacements_topic: Optional[str]
+    replacements_topic_spec = stream_loader.get_replacement_topic_spec()
+
+    if replacements_topic_spec is None:
+        if replacements_topic is not None:
+            raise ValueError("Commit log not supported for this storage")
+
+        resolved_replacements_topic = None
+    elif commit_log_topic is not None:
+        resolved_commit_log_topic = replacements_topic
+    else:
+        resolved_commit_log_topic = replacements_topic_spec.get_physical_topic_name(
+            slice_id
+        )
+
+    # TODO: Temporarily hardcoded. To be properly resolved based on the storage set.
+    hardcoded_clickhouse_cluster = ClickhouseClusterConfig(
+        host="127.0.0.1", port=9000, user="default", password="", database="default"
+    )
+
+    return RustConsumerConfig(
+        storages=[
+            StorageConfig(
+                name=storage_name, clickhouse_cluster=hardcoded_clickhouse_cluster
```

Comment:
can you resolve table here as well? or should I continue parsing yaml

Reply:
Done!